koku/reporting_common/models.py

```diff
@@ -141,3 +151,84 @@ class Meta:
 
     region = models.CharField(max_length=32, null=False, unique=True)
     region_name = models.CharField(max_length=64, null=False, unique=True)
+
+
+class DelayedCeleryTasks(models.Model):
+    """This table tracks summary tasks that are delaying execution."""
+
+    class Meta:
+        db_table = "delayed_celery_tasks"
+
+    task_name = models.CharField(max_length=255)
+    task_args = models.TextField()
+    task_kwargs = models.TextField()
+    timeout_timestamp = models.DateTimeField()
+    provider_uuid = models.UUIDField()
+    queue_name = models.CharField(max_length=255)
+    metadata = models.JSONField(default=dict, null=True)
+
+    def set_task_args(self, args):
+        self.task_args = json.dumps(args)
+
+    def get_task_args(self):
+        return json.loads(self.task_args)
+
+    def set_task_kwargs(self, kwargs):
+        self.task_kwargs = json.dumps(kwargs)
+
+    def get_task_kwargs(self):
+        return json.loads(self.task_kwargs)
```

Comment:
Are these methods necessary if the task_args and task_kwargs fields are JSONFields instead? I see json.load/dump in several places around these fields, so it seems like they should be JSONFields.

Reply:
Let me find out... 🤔

Reply:
You are correct! Awesome suggestion @maskarb Thank you.